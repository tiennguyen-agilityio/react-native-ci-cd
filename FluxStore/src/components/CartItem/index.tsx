import {memo, useMemo} from 'react';
import {StyleSheet} from 'react-native';
import isEqual from 'react-fast-compare';
import FastImage from 'react-native-fast-image';

// Interfaces
import {Cart} from '@/interfaces';

// Constants
import {CURRENCY_UNIT} from '@/constants';

// Themes
import {borderRadius, fontSizes, colors as TColors} from '@/themes';

// Stores
import {useThemeStore} from '@/stores';

// Components
import {CheckBoxIcon} from '../Icons';
import Quantity from '../Quantity';
import Flex from '../Flex';
import Text from '../Text';

interface CartItemProps extends Cart {
  isChecked: boolean;
  onChangeChecked: () => void;
  onChangeQuantity: (quantity: number) => void;
}

const CartItem = ({
  product,
  quantity,
  isChecked,
  sizes,
  colors,
  onChangeChecked,
  onChangeQuantity,
}: CartItemProps) => {
  const {
    theme: {background, fonts},
  } = useThemeStore();
  const {name = '', image = '', price = ''} = product || {};

  const styles = useMemo(
    () =>
      StyleSheet.create({
        wrapper: {
          height: 99,
          flexDirection: 'row',
          borderRadius: borderRadius.lg,
          shadowColor: TColors.gray[500],
          shadowOffset: {width: 0, height: 1},
          shadowOpacity: 0.1,
          shadowRadius: borderRadius.lg,
          elevation: 3,
          backgroundColor: background.default,
        },
        image: {
          width: 99,
          height: '100%',
          borderTopLeftRadius: borderRadius.lg,
          borderBottomLeftRadius: borderRadius.lg,
          backgroundColor: background.tertiary,
        },
        info: {
          flex: 1,
          flexDirection: 'row',
          justifyContent: 'space-between',
          height: '100%',
          paddingTop: 16,
          paddingBottom: 13,
          paddingLeft: 13,
          paddingRight: 20,
          borderTopRightRadius: borderRadius.lg,
          borderBottomRightRadius: borderRadius.lg,
        },
        text: {
          textTransform: 'capitalize',
          fontSize: fontSizes.mini,
        },
        price: {
          fontFamily: fonts.primary.bold,
        },
      }),
    [fonts, background],
  );

  return (
    <Flex style={styles.wrapper} backgroundColor={background.default}>
      <FastImage
        style={styles.image}
        source={{
          uri: image,
          priority: FastImage.priority.normal,
        }}
        resizeMode={FastImage.resizeMode.cover}
      />
      <Flex style={styles.info}>
        <Flex flex={1} justify="between">
          <Text variant="title" fontSize={fontSizes.xxs} numberOfLines={1} ellipsizeMode="tail">
            {name}
          </Text>
          <Text style={styles.price} variant="subTitle" numberOfLines={1} ellipsizeMode="tail">
            {`${CURRENCY_UNIT} ${price}`}
          </Text>
          <Text numberOfLines={1} ellipsizeMode="tail" style={styles.text}>
            {`Size: ${sizes} | Color: ${colors}`}
          </Text>
        </Flex>
        <Flex justify="between" align="end" marginRight={0}>
          <CheckBoxIcon isActive={isChecked} onPress={onChangeChecked} />
          <Quantity defaultValue={quantity} onChangeValue={onChangeQuantity} />
        </Flex>
      </Flex>
    </Flex>
  );
};

export default memo(CartItem, isEqual);
