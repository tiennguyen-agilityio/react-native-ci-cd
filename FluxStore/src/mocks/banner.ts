export const BANNERS = [
  {
    id: '01',
    title: 'Clothing Collection 2025',
    imageUrl: 'https://i.ibb.co/wZv2s1T6/Frame-33149.png',
  },
  {
    id: '02',
    title: 'Autumn Collection 2025',
    imageUrl: 'https://i.ibb.co/hRBFHvHk/Mask-Group-1.png',
  },
  {
    id: '03',
    title: 'Accessories Collection 2025',
    imageUrl: 'https://i.ibb.co/20WCrpXW/Frame-33123.png',
  },
];
