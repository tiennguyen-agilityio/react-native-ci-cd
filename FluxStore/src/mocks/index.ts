export * from './product';
export * from './banner';
export * from './review';
export * from './cart';
