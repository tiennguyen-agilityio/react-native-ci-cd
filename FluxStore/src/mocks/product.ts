import {Product} from '@/interfaces';

export const COLORS = ['black', 'red', 'blue'];
export const SIZES = ['S', 'M', 'L'];

export const PRODUCTS: Product[] = [
  {
    id: '01',
    name: 'Linen Dress',
    price: 90.02,
    discount: 10,
    rating: 5,
    reviewCount: 20,
    image: 'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
    sizes: ['s', 'M'],
    colors: ['black', 'red', 'blue'],
    carouselImages: [
      'https://i.ibb.co/KQj6DyL/image-87.png',
      'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
      'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    ],
  },
  {
    id: '02',
    name: 'Filted Waist Dress',
    price: 82.0,
    discount: 0,
    rating: 3,
    reviewCount: 110,
    image: 'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    description:
      'Minimal Stand is made of by natural wood. The design that is very simple and minimal. This is truly one of the best furnitures in any family for now. With 3 different colors, you can easily select the best match for your home.',
    sizes: ['XL'],
    colors: ['pink', 'black'],
    carouselImages: [
      'https://i.ibb.co/KQj6DyL/image-87.png',
      'https://i.ibb.co/SD47BZNp/Mask-Group-1.png',
      'https://i.ibb.co/Jw9tjsj3/Mask-Group-2.png',
    ],
  },
];
