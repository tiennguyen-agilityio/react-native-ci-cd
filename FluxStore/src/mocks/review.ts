export const REVIEWS = [
  {
    id: '01',
    avatarUrl: 'https://i.ibb.co/h1n0FRHn/Ellipse-76.png',
    name: 'Jennifer Rose',
    rating: 5,
    comment:
      'I love it.  Awesome customer service!! Helped me out with adding an additional item to my order. Thanks again!',
    createdAt: 1747710042553,
  },
  {
    id: '02',
    avatarUrl: 'https://i.ibb.co/tT8QbLVp/Ellipse-77.png',
    name: 'Kelly Rihana',
    rating: 5,
    comment: "I'm very happy with order, It was delivered on and good quality. Recommended!",
    createdAt: 1747710042570,
  },
];
