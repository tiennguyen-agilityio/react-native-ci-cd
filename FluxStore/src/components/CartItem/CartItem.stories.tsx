import {Meta, StoryObj} from '@storybook/react';

import CartItem from '.';
import {CARTS} from '@/mocks';
import {useState} from 'react';
import {Alert} from 'react-native';

const meta: Meta<typeof CartItem> = {
  title: 'Components/CartItem',
  component: CartItem,
};

export default meta;

type Story = StoryObj<typeof CartItem>;

const defaultProps = {
  ...CARTS[0],
};

const Render = ({isChecked}: {isChecked: boolean}) => {
  const [isCheckActive, setIsCheckActive] = useState(isChecked);
  const [quantity, setQuantity] = useState(1);

  const handleChangeChecked = () => {
    setIsCheckActive(!isCheckActive);
    Alert.alert(`Handle change checked ${!isCheckActive}`);
  };

  const handleChangeQuantity = (amount: number) => {
    setQuantity(amount);
    Alert.alert(`Handle change quantity ${amount}`);
  };

  return (
    <CartItem
      {...defaultProps}
      isChecked={isCheckActive}
      quantity={quantity}
      onChangeChecked={handleChangeChecked}
      onChangeQuantity={handleChangeQuantity}
    />
  );
};

export const Primary: Story = {
  render: Render,
};

export const Secondary: Story = {
  render: () => Render({isChecked: true}),
};
