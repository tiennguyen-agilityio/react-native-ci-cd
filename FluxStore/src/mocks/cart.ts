import {PRODUCTS} from './product';

export const CARTS = [
  {
    id: '01',
    product: {
      id: PRODUCTS[0].id,
      image: PRODUCTS[0].image,
      name: PRODUCTS[0].name,
      price: PRODUCTS[0].price,
    },
    sizes: [PRODUCTS[0]?.sizes?.[0], PRODUCTS[0]?.sizes?.[1]].join(', '),
    colors: [PRODUCTS[0]?.colors?.[0]].join(', '),
    quantity: 2,
    isChecked: true,
  },
  {
    id: '02',
    product: {
      id: PRODUCTS[1].id,
      image: PRODUCTS[1].image,
      name: PRODUCTS[1].name,
      price: PRODUCTS[1].price,
    },
    sizes: [PRODUCTS[1]?.sizes?.[0]].join(', '),
    colors: [PRODUCTS[1]?.colors?.[0], PRODUCTS[1]?.colors?.[1]].join(', '),
    quantity: 1,
    isChecked: false,
  },
];
