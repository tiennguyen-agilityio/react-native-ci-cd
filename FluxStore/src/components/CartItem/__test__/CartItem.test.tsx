import {render} from '@testing-library/react-native';

import CartItem from '..';
import {CARTS} from '@/mocks';

describe('CartItem', () => {
  const onChangeChecked = jest.fn();
  const onChangeQuantity = jest.fn();

  const props = {
    ...CARTS[0],
    onChangeChecked,
    onChangeQuantity,
  };

  it('should render CartItem component', () => {
    const {toJSON} = render(<CartItem {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render CartItem with isChecked true', () => {
    const {toJSON} = render(<CartItem {...props} isChecked />);

    expect(toJSON()).toMatchSnapshot();
  });
});
