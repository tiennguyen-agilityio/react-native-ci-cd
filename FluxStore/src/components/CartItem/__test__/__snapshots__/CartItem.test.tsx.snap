// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should render CartItem component 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "alignItems": "stretch",
      },
      {
        "flexWrap": "nowrap",
      },
      {
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "elevation": 3,
        "flexDirection": "row",
        "height": 99,
        "shadowColor": "#9b9b9b",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 20,
      },
      {
        "backgroundColor": "#fff",
      },
    ]
  }
  testID="flex"
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "backgroundColor": "#f8f8fa",
          "borderBottomLeftRadius": 20,
          "borderTopLeftRadius": 20,
          "height": "100%",
          "width": 99,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "priority": "normal",
          "uri": "https://i.ibb.co/SD47BZNp/Mask-Group-1.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "flexDirection": "column",
        },
        {
          "justifyContent": "flex-start",
        },
        {
          "alignItems": "stretch",
        },
        {
          "flexWrap": "nowrap",
        },
        {
          "borderBottomRightRadius": 20,
          "borderTopRightRadius": 20,
          "flex": 1,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingBottom": 13,
          "paddingLeft": 13,
          "paddingRight": 20,
          "paddingTop": 16,
        },
        {},
      ]
    }
    testID="flex"
  >
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "space-between",
          },
          {
            "alignItems": "stretch",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "flex": 1,
          },
        ]
      }
      testID="flex"
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          [
            {
              "letterSpacing": 0,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            {
              "color": "#000",
              "fontFamily": "NunitoSemiBold",
              "fontSize": 20,
              "fontWeight": 700,
            },
            {
              "fontSize": 13,
            },
            undefined,
          ]
        }
      >
        Linen Dress
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          [
            {
              "letterSpacing": 0,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            {
              "color": "#000",
              "fontFamily": "RobotoRegular",
              "fontSize": 16,
              "fontWeight": 700,
            },
            {},
            {
              "fontFamily": "ProductSansBold",
            },
          ]
        }
      >
        $ 90.02
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          [
            {
              "letterSpacing": 0,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            {
              "color": "#737680",
              "fontFamily": "ProductSansRegular",
              "fontSize": 14,
            },
            {},
            {
              "fontSize": 10,
              "textTransform": "capitalize",
            },
          ]
        }
      >
        Size: s, M | Color: black
      </Text>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "space-between",
          },
          {
            "alignItems": "flex-end",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "marginRight": 0,
          },
        ]
      }
      testID="flex"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="checkbox-icon"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={20}
          bbWidth={19}
          fill="none"
          focusable={false}
          height={20}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 20,
                "width": 19,
              },
            ]
          }
          vbHeight={20}
          vbWidth={19}
          width={19}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGRect
              fill={
                {
                  "payload": 4283468411,
                  "type": 0,
                }
              }
              height={20}
              propList={
                [
                  "fill",
                ]
              }
              rx={4}
              width={18.2891}
              x={0.592896}
              y={0}
            />
            <RNSVGPath
              d="M5.165 9.47L8.014 13l5.381-6"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 20,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 22,
              "justifyContent": "space-between",
              "paddingHorizontal": 8,
              "width": 63,
            },
            {
              "borderColor": "#00000080",
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "height": 13,
              "justifyContent": "center",
              "opacity": 1,
              "width": 13,
            }
          }
          testID="minus-icon"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={3}
            bbWidth={6}
            fill="none"
            focusable={false}
            height={3}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 3,
                  "width": 6,
                },
              ]
            }
            vbHeight={3}
            vbWidth={6}
            width={6}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M0.100098 1.31445L5.89475 1.31445"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                  ]
                }
                stroke={
                  {
                    "payload": 2147483648,
                    "type": 0,
                  }
                }
                strokeWidth={1.5}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          style={
            [
              {
                "letterSpacing": 0,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              {
                "color": "#737680",
                "fontFamily": "ProductSansRegular",
                "fontSize": 14,
              },
              {
                "fontSize": 12,
              },
              [
                {
                  "flex": 0,
                },
                {
                  "color": "#00000080",
                },
              ],
            ]
          }
        >
          2
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "height": 13,
              "justifyContent": "center",
              "opacity": 1,
              "width": 13,
            }
          }
          testID="add-icon"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={8}
            bbWidth={8}
            fill="none"
            focusable={false}
            height={8}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 8,
                  "width": 8,
                },
              ]
            }
            vbHeight={14}
            vbWidth={14}
            width={8}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M13 8H8v5c0 .55-.45 1-1 1s-1-.45-1-1V8H1c-.55 0-1-.45-1-1s.45-1 1-1h5V1c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
                fill={
                  {
                    "payload": 2147483648,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`CartItem should render CartItem with isChecked true 1`] = `
<View
  style={
    [
      {
        "flexDirection": "column",
      },
      {
        "justifyContent": "flex-start",
      },
      {
        "alignItems": "stretch",
      },
      {
        "flexWrap": "nowrap",
      },
      {
        "backgroundColor": "#fff",
        "borderRadius": 20,
        "elevation": 3,
        "flexDirection": "row",
        "height": 99,
        "shadowColor": "#9b9b9b",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 20,
      },
      {
        "backgroundColor": "#fff",
      },
    ]
  }
  testID="flex"
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "backgroundColor": "#f8f8fa",
          "borderBottomLeftRadius": 20,
          "borderTopLeftRadius": 20,
          "height": "100%",
          "width": 99,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="cover"
      source={
        {
          "priority": "normal",
          "uri": "https://i.ibb.co/SD47BZNp/Mask-Group-1.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      [
        {
          "flexDirection": "column",
        },
        {
          "justifyContent": "flex-start",
        },
        {
          "alignItems": "stretch",
        },
        {
          "flexWrap": "nowrap",
        },
        {
          "borderBottomRightRadius": 20,
          "borderTopRightRadius": 20,
          "flex": 1,
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingBottom": 13,
          "paddingLeft": 13,
          "paddingRight": 20,
          "paddingTop": 16,
        },
        {},
      ]
    }
    testID="flex"
  >
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "space-between",
          },
          {
            "alignItems": "stretch",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "flex": 1,
          },
        ]
      }
      testID="flex"
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          [
            {
              "letterSpacing": 0,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            {
              "color": "#000",
              "fontFamily": "NunitoSemiBold",
              "fontSize": 20,
              "fontWeight": 700,
            },
            {
              "fontSize": 13,
            },
            undefined,
          ]
        }
      >
        Linen Dress
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          [
            {
              "letterSpacing": 0,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            {
              "color": "#000",
              "fontFamily": "RobotoRegular",
              "fontSize": 16,
              "fontWeight": 700,
            },
            {},
            {
              "fontFamily": "ProductSansBold",
            },
          ]
        }
      >
        $ 90.02
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          [
            {
              "letterSpacing": 0,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            {
              "color": "#737680",
              "fontFamily": "ProductSansRegular",
              "fontSize": 14,
            },
            {},
            {
              "fontSize": 10,
              "textTransform": "capitalize",
            },
          ]
        }
      >
        Size: s, M | Color: black
      </Text>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "column",
          },
          {
            "justifyContent": "space-between",
          },
          {
            "alignItems": "flex-end",
          },
          {
            "flexWrap": "nowrap",
          },
          undefined,
          {
            "marginRight": 0,
          },
        ]
      }
      testID="flex"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="checkbox-icon"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={20}
          bbWidth={19}
          fill="none"
          focusable={false}
          height={20}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 20,
                "width": 19,
              },
            ]
          }
          vbHeight={20}
          vbWidth={19}
          width={19}
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGRect
              fill={
                {
                  "payload": 4283468411,
                  "type": 0,
                }
              }
              height={20}
              propList={
                [
                  "fill",
                ]
              }
              rx={4}
              width={18.2891}
              x={0.592896}
              y={0}
            />
            <RNSVGPath
              d="M5.165 9.47L8.014 13l5.381-6"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
              propList={
                [
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "borderRadius": 20,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 22,
              "justifyContent": "space-between",
              "paddingHorizontal": 8,
              "width": 63,
            },
            {
              "borderColor": "#00000080",
            },
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "height": 13,
              "justifyContent": "center",
              "opacity": 1,
              "width": 13,
            }
          }
          testID="minus-icon"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={3}
            bbWidth={6}
            fill="none"
            focusable={false}
            height={3}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 3,
                  "width": 6,
                },
              ]
            }
            vbHeight={3}
            vbWidth={6}
            width={6}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M0.100098 1.31445L5.89475 1.31445"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                  ]
                }
                stroke={
                  {
                    "payload": 2147483648,
                    "type": 0,
                  }
                }
                strokeWidth={1.5}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          style={
            [
              {
                "letterSpacing": 0,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              {
                "color": "#737680",
                "fontFamily": "ProductSansRegular",
                "fontSize": 14,
              },
              {
                "fontSize": 12,
              },
              [
                {
                  "flex": 0,
                },
                {
                  "color": "#00000080",
                },
              ],
            ]
          }
        >
          2
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "height": 13,
              "justifyContent": "center",
              "opacity": 1,
              "width": 13,
            }
          }
          testID="add-icon"
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={8}
            bbWidth={8}
            fill="none"
            focusable={false}
            height={8}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 8,
                  "width": 8,
                },
              ]
            }
            vbHeight={14}
            vbWidth={14}
            width={8}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M13 8H8v5c0 .55-.45 1-1 1s-1-.45-1-1V8H1c-.55 0-1-.45-1-1s.45-1 1-1h5V1c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
                fill={
                  {
                    "payload": 2147483648,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;
